nxcomp/Log.h

```diff
@@ -211,12 +212,15 @@ class NXLog
 	void flush()
 	{
 		per_thread_data *pdt = get_data();
+		const std::string str = pdt->buffer->str();
```

Comment:
General comment: do we want to trim excess whitespace at the end or beginning of a string? If so, a free-standing utility function (or multiple - probably not in the namespace of the logger but a more general utility namespace) that does the equivalent of

```C++
  // front-based trimming
  foo.erase(foo.begin(), std::find_if(foo.begin(), foo.end(), std::not1(std::ptr_fun<int, int>(std::isspace))));
  // back-based trimming
  foo.erase(std::find_if(foo.rbegin(), foo.rend(), std::not1(std::ptr_fun<int, int>(std::isspace))).base(), foo.end());
```

We need to be careful about std::isspace though, because we may not want to drop additional newlines at the end of the string. Or even at the beginning.